standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/properties/HMSServletTest.java

```diff
@@ -0,0 +1,278 @@
+        .build();
+    Date now = new Date();
+    Date expirationTime = new Date(now.getTime() + lifeTimeMillis);
+    JWTClaimsSet claimsSet = new JWTClaimsSet.Builder()
+        .jwtID(UUID.randomUUID().toString())
+        .issueTime(now)
+        .issuer("auth-server")
+        .subject(user)
+        .expirationTime(expirationTime)
+        .claim("custom-claim-or-payload", "custom-claim-or-payload")
+        .build();
+    SignedJWT signedJWT = new SignedJWT(header, claimsSet);
+    // Compute the RSA signature
+    signedJWT.sign(signer);
+    return signedJWT.serialize();
+  }
+
+
+  /**
+   * A property client that uses http as transport.
+   */
+  public static class JSonClient implements PropertyClient {
+    private final URL url;
+    private String jwt = null;
+    JSonClient(String token, URL url) {
+      this.jwt = token;
+      this.url = url;
+    }
+
+    public boolean setProperties(Map<String, String> properties) {
+      try {
+        clientCall(jwt, url, "PUT", properties);
+        return true;
+      } catch(IOException xio) {
+        return false;
+      }
+    }
+
+    public Map<String, Map<String, String>> getProperties(String mapPrefix, String mapPredicate, String... selection) throws IOException {
+      Map<String, Object> args = new TreeMap<>();
+      args.put("prefix", mapPrefix);
+      if (mapPredicate != null) {
+        args.put("predicate", mapPredicate);
+      }
+      if (selection != null && selection.length > 0) {
+        args.put("selection", selection);
+      }
+      try {
+        Object result = clientCall(jwt, url, "POST", args);
+        return result instanceof Map? (Map<String, Map<String, String>>) result : null ;
+      } catch(IOException xio) {
+        return null;
+      }
+    }
+  }
+
+  @Test
+  public void testJSONServlet() throws Exception {
+    URL url = new URL("http://hive@localhost:" + sport + "/" + CLI + "/" + NS);
+    Map<String, String> json = Collections.singletonMap("method", "echo");
+    String jwt = generateJWT();
+    Object response = clientCall(jwt, url, "POST", json);
+    Assert.assertNotNull(response);
+    Assert.assertEquals(json, response);
+
+    response = clientCall(null, url, "POST", json);
+    Assert.assertNull(response);
+  }
+
+  @Test
+  public void testProperties1() throws Exception {
+      runOtherProperties1(client);
+
+  }
+
+  @Test
+  public void testProperties0() throws Exception {
+      runOtherProperties0(client);
+  }
+
+  @Test
+  public void testPropertiesOtherClient() throws Exception {
+    HttpClient client = new HttpClient();
+    HttpMethod method = new PostMethod("http://hive@localhost:" + sport + "/" + CLI + "/" + NS);
+
+    String jwt = generateJWT();
+    method.addRequestHeader("Authorization","Bearer " + jwt);
+    method.addRequestHeader("Content-Type", "application/json");
+    method.addRequestHeader("Accept", "application/json");
+
+    String msgBody = "{\"method\":\"echo\"}";
```

Comment:
nit: can we add a unit test for method `selectProperties`

Reply:
Added.